src/main/java/com/example/urlmanagement/encoder/ShortUrlEncoder.java

```diff
@@ -0,0 +1,6 @@
+package com.example.urlmanagement.encoder;
+
+public interface ShortUrlEncoder {
+
+    String createShortUrl(Long resource);
+}
```

Comment:
Mapper를 별도로 두는 것도 방법일 수 있으나 아래처럼 인코딩 타입을 지원하는지 체크하여 서비스에서 List<ShortUrlEncoder>를 주입받아 적절한 인코더를 꺼내와서 사용하는 방법도 고려해볼 수 있어요.
이렇게 되면 별도로 꺼내는 로직이 분산되면서 관리되지 않다는 장점이 있을 것 같아요.
```suggestion
public interface ShortUrlEncoder {
    boolean supports(EncodeType encodeType); 
    String createShortUrl(Long resource);
}
```